### DISCUSSION TITLE
issues on getting started

### DISCUSSION THREAD

**Discussion by ghost** [bevyengine/bevy#1232] [Q&A]
Hi, I am new to Bevy and I was following the [tutorial](https://bevyengine.org/learn/book/getting-started/plugins/),
but it seems like something's wrong on my side after adding the DefaultPlugins,

for some reason, on my end, the code doesn't loop as it is said it should, it just runs once and then exits with 0,
I don't see where this could be coming from

my code :
```rust
use bevy::prelude::*;

struct Person;
struct Name(String);

fn add_people(commands: &mut Commands) {
    commands
        .spawn((Person, Name("Person 1".to_string())))
        .spawn((Person, Name("Person 2".to_string())))
        .spawn((Person, Name("Person 3".to_string())));
}

struct GreetTimer(Timer);

fn greet_people(query: Query<&Name, With<Person>>) {
    for name in query.iter() {
        println!("Hello {}!", name.0);
    }
}

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_system(add_people.system())
        .add_system(greet_people.system())
        .run();
}
```

**Comment by TheRawMeatball** [comment]
Can you paste the full output, and the output of `cargo version`? This works for me

**Comment by ghost** [comment]
```
/home/infrandomness/.cargo/bin/cargo run --color=always --package game --bin game
warning: struct is never constructed: `GreetTimer`
  --> src/main.rs:13:8
   |
13 | struct GreetTimer(Timer);
   |        ^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/game`
Hello Person 1!
Hello Person 2!
Hello Person 3!

Process finished with exit code 0
```

cargo 1.51.0-nightly (329895f5b 2021-01-06)

**Reply by ghost** [reply]
don't mind the warning

**Comment by TheRawMeatball** [comment]
can you try compiling with stable?

**Comment by ghost** [comment]
It does the same thing on stable without "-Zshare-generics=y" in the cargo configuration file

**Comment by ghost** [comment]
Here's a bit more infos on my environment : 
os / distro : linux / OpenSUSE Leap 15.2
kernel : 5.3.18-lp152.57-default
rustc : nightly-2021-01-10
IDE : Clion 2020.3
Bevy : v0.4

**Comment by ghost** [comment]
can anyone help me ?

**Reply by cart** [reply]
Hmm can you successfully run the wgpu examples?
https://github.com/gfx-rs/wgpu-rs

**Reply by ghost** [reply]
Hi, I tried the cube example from wgpu-rs, first I was missing the c++ executable which I create using
```
sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 3000
```

after it finished compiling, I got that error message : 
```
thread 'main' panicked at 'No suitable GPU adapters found on the system!', examples/cube/../framework.rs:145:10
stack backtrace:
   0: rust_begin_unwind
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:495:5
   1: core::panicking::panic_fmt
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/panicking.rs:92:14
   2: core::option::expect_failed
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/option.rs:1260:5
   3: core::option::Option<T>::expect
             at /home/infrandomness/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:349:21
   4: cube::framework::setup::{{closure}}
             at ./examples/cube/../framework.rs:139:19
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /home/infrandomness/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:80:19
   6: pollster::block_on
             at /home/infrandomness/.cargo/registry/src/github.com-1ecc6299db9ec823/pollster-0.2.0/src/lib.rs:64:24
   7: cube::framework::run
             at ./examples/cube/../framework.rs:337:17
   8: cube::main
             at ./examples/cube/main.rs:428:5
   9: core::ops::function::FnOnce::call_once
             at /home/infrandomness/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

I run on no dgpu, there's only one igpu in the computer I had this issue with wgpu-rs and the issue with bevy

**Reply by ghost** [reply]
I have tried the same thing on my desktop computer running a gtx 1060 6gb with nouveau and openSUSE Leap 15.2 too, the same issue occurs there
Maybe I am missing something on both systems ?

**Reply by cart** [reply]
hmm looks like the nouveau driver might not support vulkan yet? https://www.phoronix.com/scan.php?page=news_item&px=Nouveau-2020-Early-Status. I know proprietary drivers are a deal breaker for some people, but I have a gtx 1070 and it works as expected on linux with nvidia's official driver.  

What is the integrated gpu you're using on the other system? I would first check to see if it supports vulkan. If it does, then you are probably missing a vulkan-related package. I'm not familiar with openSUSE, but you should check their relevant support forums / wikis to identify the package required.

**Reply by ghost** [reply]
I use an intel igpu, the drivers are a module of the kernel so it works straight out of the box,

I know this one supports vulkan, I've already written software with https://github.com/vulkano-rs/vulkano that used this igpu
I'm missing something vulkan related

**Reply by ghost** [reply]
Okay I was missing Mesa-libVulkan-devel and probably also the vulkan loaders, now the cube example works

However the same problem still happens in my original game with the piece of code I sent up

**Reply by cart** [reply]
Hmm this is just a shot in the dark: maybe it is wayland related? we don't enable native wayland by default to keep dependency counts low (wayland requires that you draw your own window decorations). Normally this isn't a problem because xwayland is installed. Can you:

1. check if you are running wayland
2. if yes, either install xwayland if it isn't already (honestly if you're getting by without xwayland im very surprised) or compile Bevy with the `wayland` feature flag. Ex: ```cargo run --example 3d_scene --features wayland```

**Reply by ghost** [reply]
I don't use wayland, I use xorg and xfce4

**Reply by cart** [reply]
To be clear, before you fixed the "vulkan driver" problem, Bevy still ran your systems once without any error messages when you had DefaultPlugins registered? If so, did a window pop up?

**Reply by ghost** [reply]
No, I had no error at all and no window popping up at all, just that it ran once instead of in a loop as it should've

**Reply by cart** [reply]
That tells us a couple of things:

1. The WinitPlugin either failed to start up or wasn't registered
2. The WgpuPlugin never initialized an adapter, otherwise it would have failed

Are you disabling any features when you build bevy? Can you zip up your project and upload it here?

**Reply by ghost** [reply]
... I forgot to remove "default-features = false" from the Cargo.toml file ...

I've just retried with every features and now it works as intended
I feel really dumb, thanks !
